Incorrect variable used

https://github.com/scylladb/seastar/blob/745cec5ea4e13f9afdb7b9596b9b8954585033b0/src/core/smp.cc#L219

This gives the error:
```
ninja: job failed: /usr/bin/g++ -DFMT_SHARED -DSEASTAR_API_LEVEL=7 -DSEASTAR_BUILD_SHARED_LIBS -DSEASTAR_DEFERRED_ACTION_REQUIRE_NOEXCEPT -DSEASTAR_HAS_MEMBARRIER -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_URING -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -Dseastar_EXPORTS -I/builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/include -I/builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/build/gen/include -I/builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/src -Os -fstack-clash-protection -Wformat -Werror=format-security -D_GLIBCXX_ASSERTIONS=1 -DSEASTAR_DEFAULT_ALLOCATOR -Os -DNDEBUG -std=gnu++20 -fPIC -U_FORTIFY_SOURCE -Wno-maybe-uninitialized -DSEASTAR_SSTRING -Wno-error=unused-result -UNDEBUG -Wall -Wimplicit-fallthrough -Wno-array-bounds -Wno-error=deprecated-declarations -gz -MD -MT CMakeFiles/seastar.dir/src/core/smp.cc.o -MF CMakeFiles/seastar.dir/src/core/smp.cc.o.d -o CMakeFiles/seastar.dir/src/core/smp.cc.o -c /builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/src/core/smp.cc
/builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/src/core/smp.cc: In lambda function:
/builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/src/core/smp.cc:219:12: error: request for member 'fetch_or' in '* p', which is of non-class type 'char'
  219 |         p->fetch_or(0, std::memory_order_relaxed);
      |            ^~~~~~~~
/builds/a16bitsysop/aports/testing/seastar/src/seastar-4f8c225d48565f5df4f034dcb55f5b5d2038c362/src/core/smp.cc:218:14: warning: unused variable 'p1' [-Wunused-variable]
  218 |         auto p1 = reinterpret_cast<volatile std::atomic<char>*>(p);
      |              ^~
ninja: subcommand failed
```

if changed to `p1->fetch_or(0, std::memory_order_relaxed);` seastar compiles on ppc64 and s390x

--- a/src/core/smp.cc
+++ b/src/core/smp.cc
@@ -216,7 +216,7 @@ internal::memory_prefaulter::work(std::vector<memory::internal::memory_range>& r
 #else
         // atomic_ref would be better, but alas C++20 only
         auto p1 = reinterpret_cast<volatile std::atomic<char>*>(p);
-        p->fetch_or(0, std::memory_order_relaxed);
+        p1->fetch_or(0, std::memory_order_relaxed);
 #endif
     };
     while (!_stop_request.load(std::memory_order_relaxed) && !ranges.empty()) {

